src/Schema/AbstractSchemaManager.php

```diff
@@ -667,7 +669,11 @@ protected function _getPortableTableColumnList(string $table, string $database,
     {
         $list = [];
         foreach ($tableColumns as $tableColumn) {
-            $column = $this->_getPortableTableColumnDefinition($tableColumn);
+            try {
+                $column = $this->_getPortableTableColumnDefinition($tableColumn);
+            } catch (UnknownColumnType $unknownTypeException) {
+                throw UnknownColumnType::withContext($unknownTypeException->getType(), sprintf('table %s', $table));
```

Comment:
I think this is overcomplication. Instead, we could reusing `UnsupportedTableDefinition` and adding another static constructor (e.g. `fromIntrospectionException()`. It would accept the table name and reference the exception caught from `_getPortableTableColumnDefinition()` as the previous exception.

This way, we will be able to supply any introspection exception with the context (table name) w/o introducing new classes.

Reply:
Sounds like a reasonable idea but what is this `UnsupportedTableDefinition` class? I can't find it in the source code or the dependencies.

Reply:
Hm thinking about this, this change might be better suited for 5.0 anyway then seeing that using a different exception would also change the type of exception thrown in the method. However, I think it still would be cool to have it in 4.3.x without any bc breaks.

How do other parts of the application handle those kinds of things. Like adding context to an exception sounds like a common enough use-case.